src/source/PeerConnection/SessionDescription.c

```diff
@@ -316,6 +327,64 @@ PCHAR fmtpForPayloadType(UINT64 payloadType, PSessionDescription pSessionDescrip
     return NULL;
 }
 
+/*
+ * Extracts a (hex) value after the provided prefix string. Returns true if
+ * successful.
+ */
+static BOOL readHexValue(const PCHAR input, const PCHAR prefix, PINT32 value) {
+  const PCHAR substr = STRSTR(input, prefix);
+  if (substr != NULL) {
+    if (sscanf(substr + STRLEN(prefix), "%x", value) == 1) {
+      return TRUE;
+    }
+  }
+  return FALSE;
+}
+
+/*
+ * Scores the provided fmtp string based on this library's ability to
+ * process various types of H264 streams. A score of 0 indicates an
+ * incompatible fmtp line. Beyond this, a higher score indicates more
+ * compatibility with the desired characteristics, packetization-mode=1,
+ * level-asymmetry-allowed=1, and inbound match with our preferred
+ * encoding level.
+ */
+INT32 getH264FmtpScore(PCHAR fmtp) {
+  INT32 profileId = 0, packetizationMode = 0, levelAsymmetry = 0;
+  BOOL isAsymmetryAllowed = FALSE;
+  BOOL isProfileMatch = FALSE;
+  INT32 score = 0;
+
+  // No ftmp match found.
+  if (fmtp == NULL) {
+      return 0;
+  }
+
+  // Currently, the packetization mode must be 1, as the packetization logic
+  // is currently not configurable, and sends both NALU and FU-A packets.
+  // https://tools.ietf.org/html/rfc7742#section-6.2
+  if (readHexValue(fmtp, "packetization-mode=", &packetizationMode) == FALSE || packetizationMode != 1) {
+      return 0;
+  }
```

Comment:
Note that this may be TOO stringent in cases where we are in rcvonly mode, as technically, the depayloader will handle packetization modes 0 and 1. However, in sndonly and sndrecv cases, there isn't any logic to protect the incoming packets from exceeding the MTU, triggering the creation of an FU-A type packet, which is only supported in packetization-mode=1. 

(Based on the packet types I saw, packetization-mode=2 does not appear to be fully implemented/supported, nor is it likely desirable in cases where you are real-time streaming?)

I'm a bit torn on this implementation, and could see making `packetization-mode` matching softly scored, but ideally that would only happen if the transceiver was only configure to receive and not send.

Reply:
I agree. We really need to look deeper at whether we need to add support for interleaved mode with MTAP or we don't need to. Based on how widespread the support is